homeassistant/components/onvif/base.py

```diff
@@ -22,10 +23,17 @@ def available(self):
     @property
     def device_info(self):
         """Return a device description for device registry."""
-        return {
-            "connections": {(CONNECTION_NETWORK_MAC, self.device.info.mac)},
+        device_info = {
+            "identifiers": {(DOMAIN, self.device.info.serial_number)},
             "manufacturer": self.device.info.manufacturer,
             "model": self.device.info.model,
             "name": self.device.name,
             "sw_version": self.device.info.fw_version,
         }
+
+        if self.device.info.mac:
+            device_info["connections"] = {
```

Comment:
We don't need both connections and identifiers. One is enough.

Reply:
There won't be any MAC for #35883, which is why I added identifiers back in.

Reply:
If would be sure that serial number is always available it would be enough with that identifier.

Otherwise, I think this is ok. But please add a comment why it's here since it's not in line with our standard.

Reply:
Even if I could guarantee the `SerialNumber` would always be populated and unique but sometimes I could also retrive a `MAC` from the device, I would still want to include the MAC address.  This way, the device registry can link devices from multiple integrations together.  The perfect example of this is how this ONVIF device merges with a device registry entry created by the UniFi integration.  This allows me to see all related entities together.

On a side note, you can find an example of both being populated in the developer documentation here: https://developers.home-assistant.io/docs/device_registry_index/#defining-devices

Reply:
The above can be a different conversation, however.  I've modified the code to reflect the uncertainty that serial number will always exist, but due to config flow constraints there will be at least one of the two.  I also added a detailed comment in the code.